pybedlite/overlap_detector.py

```diff
@@ -56,6 +61,52 @@
 from pybedlite.bed_source import BedSource
 
 
+class _Span(Protocol):
+    """A span with a start and an end. 0-based open-ended."""
+
+    @property
+    def start(self) -> int:
+        """A 0-based start position."""
+
+    @property
+    def end(self) -> int:
+        """A 0-based open-ended position."""
+
+
+class _GenomicSpanWithChrom(_Span, Hashable, Protocol):
+    """A genomic feature where reference sequence is accessed with `chrom`."""
+
+    @property
+    def chrom(self) -> str:
+        """A reference sequence name."""
+
+
+class _GenomicSpanWithContig(_Span, Hashable, Protocol):
+    """A genomic feature where reference sequence is accessed with `contig`."""
+
+    @property
+    def contig(self) -> str:
+        """A reference sequence name."""
+
+
+class _GenomicSpanWithRefName(_Span, Hashable, Protocol):
+    """A genomic feature where reference sequence is accessed with `refname`."""
+
+    @property
+    def refname(self) -> str:
+        """A reference sequence name."""
```

Comment:
These are private [structural types](https://docs.python.org/3/library/typing.html#typing.Protocol). I intentionally did not make them public or [runtime checkable](https://docs.python.org/3/library/typing.html#typing.runtime_checkable).